### PR TITLE
Make three levels of severity for validation messages

### PR DESCRIPTION
**Pull request by alexey-zakharenkov**
Add 'notice' severity level for validation messages. From now on the validation message levels are:

- **error** — the network is invalid and would not be added to the routing json file defined by `-o` option of `scripts/process_subways.py` script. Examples: wrong station count, stop_position not connected to a station.
- **warning** — there is an error in data but it's still possible to build routes. Examples: hole in route rails, an under construction rail in route.
- **notice** — suspicious condition but not necessarily an error. Example: route has different colour from master.

### CODE REVIEW

**Comment by claudiush** [comment]
I ran through validating a few cities locally and it looks good generally. Helpful differentiation on how to prioritize fixing items.
Some improvement items I would suggest for this PR:
- There's currently nothing in the UI that conveys the important information you captured about the different levels of validation messages in your [first message in this conversation](https://github.com/alexey-zakharenkov/subways/pull/14#issue-1027535741). Maybe preceed every section with an icon and a tooltip? Let me know if you want me to look at that in a follow up PR.
- Contrast for red colour of .errors class is not accessible. Use #ED0000 instead of #FF0000 for the red. See https://webaim.org/resources/contrastchecker/?fcolor=ED0000&bcolor=FFFFFF
- I always mistook "e: 17" in the entrances column as meaning "Number of errors". Maybe change the "e" to "ent" for entrances to clarify that

File: v2h_templates.py [hunk 1]
```diff
@@ -43,6 +49,57 @@
...
+}
+.tooltip div {
+    display: inline-block;
+    width: 19px;
+}
+.tooltip:before {
+    content: attr(data-text);
+    position: absolute;
+    top: 100%;
+    transform: translateX(-50%);
```
Comment (rostixman) [inline v2h_templates.py:71]: This property moves tooltip out of main container, when window width is less then 890px

```suggestion
    + left: 0;
```

File: v2h_templates.py [hunk 2]
```diff
@@ -43,6 +49,57 @@
...
+    transition: .3s opacity;
+    visibility: hidden;
+    z-index: 10
+}
+.tooltip:after {
+    content: "";
+    position: absolute;
+    margin-top: -5px;
+    top: 100%;
+    transform: translateX(-50%);
```
Comment (rostixman) [inline v2h_templates.py:89]:
```suggestion
    + left: 30px;
```

File: v2h_templates.py (hunk 1 above)
Comment (rostixman) [inline v2h_templates.py:71]: This property moves tooltip out of main container, when window width is less then 890px

```suggestion
    + left: 0;
```

**Review by rostixman — CHANGES_REQUESTED** [review]

File: v2h_templates.py (hunk 2 above)
Comment (alexey-zakharenkov) [reply v2h_templates.py]: done

File: v2h_templates.py (hunk 1 above)
Comment (alexey-zakharenkov) [reply v2h_templates.py]: done

File: subway_structure.py [hunk 3]
```diff
@@ -1741,8 +1748,7 @@ def validate_overground_lines(self):
             ]
         )
         if self.found_tram_lines != self.num_tram_lines:
-            self.error_if(
-                self.found_tram_lines == 0,
+            self.error(
```
Comment (Zverik) [inline subway_structure.py:1751]: This might raise errors when there are tram lines not in list.

File: subway_structure.py [hunk 4]
```diff
@@ -1780,36 +1786,40 @@ def validate(self):
...
                 )
-                self.error_if(
-                    not (
-                        0
-                        <= (self.num_stations - self.found_stations)
-                        / self.num_stations
-                        <= ALLOWED_STATIONS_MISMATCH
-                    ),
-                    msg,
+                log_function = (
```
Comment (Zverik) [inline subway_structure.py:1789]: Why not put this tructure into `error_if`?

**Review by Zverik — APPROVED** [review]
I trust you know what you're doing :)

File: subway_structure.py (hunk 4 above)
Comment (alexey-zakharenkov) [reply subway_structure.py:1789]: With three error levels more combinations arise. Now at least `error_or_warning_if` and `error_or_notice_if` functions would be needed. I removed `error_if` function at all.

File: subway_structure.py (hunk 3 above)
Comment (alexey-zakharenkov) [reply subway_structure.py]: Returned "error-if" logic in this case.

**Review by rostixman — APPROVED** [review]

**Pull request merged**